pkg/appmgmt/general/general.go

```diff
@@ -104,9 +104,15 @@ func isStateAwareDisabled(pod *v1.Pod) bool {
 	return result
 }
 
-func getOwnerReferences(pod *v1.Pod) []metav1.OwnerReference {
+func getOwnerReference(pod *v1.Pod) []metav1.OwnerReference {
 	if len(pod.OwnerReferences) > 0 {
-		return pod.OwnerReferences
+		// Filter the pod kind owner reference
+		// keep this because we need to find the original pod
+		for _, ref := range pod.OwnerReferences {
+			if ref.Kind == reflect.TypeOf(v1.Pod{}).Name() {
+				return []metav1.OwnerReference{ref}
+			}
+		}
```

Comment:
This is not needed, the call should only ever be made with the originator or owner pod.
Just always build a new reference without doing anything else.

NIT simplification:
`v1.SchemeGroupVersion.String()` is always "v1" unless we change the signature. 
`reflect.TypeOf(v1.Pod{}).Name()` is always "Pod" nothing else.

Reply:
Thank you @wilfred-s for review, addesseed in latest PR!